hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/RemoteWithExtrasException.java

```diff
@@ -98,6 +111,11 @@ private IOException instantiateException(Class<? extends IOException> cls) throw
     cn.setAccessible(true);
     IOException ex = cn.newInstance(this.getMessage());
     ex.initCause(this);
+
+    if (ex instanceof HBaseServerException) {
```

Comment:
This should be safe for existing users of `hbase.client.pause.cqtbe` as long as they upgrade the server first, then the client. If they don't, the boolean value passed in above will be false and we might erroneously downgrade a CQTBE to !isServerOverloaded. 

I'm going to look into tightening this up since I think we should honor the fact that the CQTBE constructor currently always passes `true` for isServerOverloaded.